### PR TITLE
Fix ruler lines rendering on top of text

### LINKED ISSUE #64704: Ruler tool line displays above text
### Godot version

0c5f254

### System information

Windows 10 x64

### Issue description

image
Probably wrong order of drawing.

### Steps to reproduce

1. Use ruler tool (R) in 2D editor
2. Looks very closely when line and text overlap

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by AlexHCC**
Fix https://github.com/godotengine/godot/issues/64704
I've changed the drawing order to draw the lines and angles first and then the text.

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -2957,6 +2957,11 @@ void CanvasItemEditor::_draw_ruler_tool() {
 		Point2 corner = Point2(begin.x, end.y);
 		Vector2 length_vector = (begin - end).abs() / zoom;
 
+		const real_t horizontal_angle_rad = length_vector.angle();
+		const real_t vertical_angle_rad = Math_PI / 2.0 - horizontal_angle_rad;
+		const int horizontal_angle = round(180 * horizontal_angle_rad / Math_PI);
+		const int vertical_angle = round(180 * vertical_angle_rad / Math_PI);
```
Comment (KoBeWi) [inline editor/plugins/canvas_item_editor_plugin.cpp:2963]: Why was this moved out of `draw_secondary_lines` condition? None of these variables are used outside.

Comment (AlexHCC) [reply editor/plugins/canvas_item_editor_plugin.cpp:2963]: There are two  `draw_secondary_lines`  if statements.  The first one to draw the lines, and the second one for the text. But `horizontal_angle_rad` and `vertical_angle_rad` are used to calculate `horizontal_angle` and `vertical_angle`, so those need to stay in the wider scope I think. But I can move `horizontal_angle` and `vertical_angle` under the second if statement.

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 2]
```diff
@@ -2974,6 +2977,42 @@ void CanvasItemEditor::_draw_ruler_tool() {
 		text_pos.x = CLAMP(text_pos.x, text_width / 2, viewport->get_rect().size.x - text_width * 1.5);
 		text_pos.y = CLAMP(text_pos.y, text_height * 1.5, viewport->get_rect().size.y - text_height * 1.5);
 
+		// Draw lines
```
Comment (KoBeWi) [inline editor/plugins/canvas_item_editor_plugin.cpp:2980]:
```suggestion
		// Draw lines.
```
Same in other comments.
(I know it was like this originally, but you could as well fix it if they are being moved)

**Review by KoBeWi — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by AlexHCC** [comment]
Thanks! 
I just want to mention that something similar can be done to `3.x`

**Pull request merged**